pkg/apis/authorization/validation/validation.go

```diff
@@ -99,3 +103,80 @@ func ValidateLocalSubjectAccessReview(sar *authorizationapi.LocalSubjectAccessRe
+
+	return allErrs
+}
+
+func validateFieldSelectorAttributes(selector *authorizationapi.FieldSelectorAttributes, fldPath *field.Path) field.ErrorList {
+	if selector == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+
+	if len(selector.RawSelector) > 0 && len(selector.Requirements) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("rawSelector"), selector.RawSelector, "may not specified at the same time as requirements"))
+	}
+	if len(selector.RawSelector) == 0 && len(selector.Requirements) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("requirements"), fmt.Sprintf("when %s is specified, requirements or rawSelector is required", fldPath)))
+	}
+
+	for i, requirement := range selector.Requirements {
+		allErrs = append(allErrs, validateFieldSelectorRequirement(requirement, fldPath.Child("requirements").Index(i))...)
+	}
+
+	return allErrs
+}
+
+func validateFieldSelectorRequirement(requirement authorizationapi.FieldSelectorRequirement, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(requirement.Key) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("key"), "must be specified"))
+	}
+
+	switch requirement.Operator {
+	case metav1.LabelSelectorOpIn, metav1.LabelSelectorOpNotIn:
+		if len(requirement.Values) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("values"), "must be specified when `operator` is 'In' or 'NotIn'"))
+		}
+	case metav1.LabelSelectorOpExists, metav1.LabelSelectorOpDoesNotExist:
+		if len(requirement.Values) > 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("values"), "may not be specified when `operator` is 'Exists' or 'DoesNotExist'"))
+		}
+	default:
+		// this is unrecognized, but we don't hard fail.  Instead the consuming code will skip evaluating these requirements.  Since requirements
+		// are all AND'd, this is safe since the requirement checked will be broader (covers) the requested requirements.
+	}
+
+	return allErrs
+}
+
+func validateLabelSelectorAttributes(selector *authorizationapi.LabelSelectorAttributes, fldPath *field.Path) field.ErrorList {
+	if selector == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+
+	if len(selector.RawSelector) > 0 && len(selector.Requirements) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("rawSelector"), selector.RawSelector, "may not specified at the same time as requirements"))
+	}
+	if len(selector.RawSelector) == 0 && len(selector.Requirements) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("requirements"), fmt.Sprintf("when %s is specified, requirements or rawSelector is required", fldPath)))
+	}
+
+	validationOptions := metav1validation.LabelSelectorValidationOptions{AllowUnknownOperatorInRequirement: true}
+	for i, requirement := range selector.Requirements {
+		allErrs = append(allErrs, metav1validation.ValidateLabelSelectorRequirement(requirement, validationOptions, fldPath.Child("requirements").Index(i))...)
```

Comment:
metav1validation.ValidateLabelSelectorRequirement fails on unrecognized operators, not sure that's what we want

Reply:
oops, missed that you added an option to tolerate that